x-pack/plugins/infra/public/components/asset_details/tabs/overview/metadata_summary.tsx

```diff
@@ -0,0 +1,117 @@
+import type { InfraMetadata } from '../../../../../common/http_api';
+import { NOT_AVAILABLE_LABEL } from '../../translations';
+import { ExpandableContent } from '../metadata/table';
+import { useTabSwitcherContext } from '../../hooks/use_tab_switcher';
+import { FlyoutTabIds, type TabIds } from '../../types';
+
+const columnTitles = {
+  hostIp: i18n.translate('xpack.infra.assetDetailsEmbeddable.overview.metadataHostIpHeading', {
+    defaultMessage: 'Host IP',
+  }),
+  hostOsVersion: i18n.translate(
+    'xpack.infra.assetDetailsEmbeddable.overview.metadataHostOsVersionHeading',
+    {
+      defaultMessage: 'Host os version',
+    }
+  ),
+};
+
+type MetadataFields = 'hostIp' | 'hostOsVersion';
+
+const metadataData = (metadataInfo: InfraMetadata['info']) => [
+  {
+    field: 'hostIp',
+    value: metadataInfo?.host?.ip,
+  },
+  {
+    field: 'hostOsVersion',
+    value: metadataInfo?.host?.os?.version,
+  },
+];
+
+interface MetadataSummaryProps {
+  metadata: InfraMetadata | null;
+  metadataLoading: boolean;
+  onShowAllClick?: (tabId: TabIds) => void;
+}
+
+export const MetadataSummary = ({
+  metadata,
+  metadataLoading,
+  onShowAllClick,
+}: MetadataSummaryProps) => {
+  const { showTab } = useTabSwitcherContext();
+
+  const onClick = () => {
+    if (onShowAllClick) {
+      onShowAllClick(FlyoutTabIds.METADATA);
+    }
+    showTab(FlyoutTabIds.METADATA);
+  };
+
+  return (
+    <>
+      <EuiFlexGroup gutterSize="m" responsive={false} wrap={true} justifyContent="spaceBetween">
+        <EuiFlexGroup alignItems="flexStart">
+          {metadataData(metadata?.info).map((metadataValue) => (
+            <EuiFlexItem key={metadataValue.field}>
+              <EuiDescriptionList data-test-subj="infraMetadataSummaryItem" compressed>
+                <EuiDescriptionListTitle
+                  css={css`
+                    white-space: nowrap;
+                  `}
+                >
+                  {columnTitles[metadataValue.field as MetadataFields]}
+                </EuiDescriptionListTitle>
+                <EuiDescriptionListDescription>
+                  {metadataLoading ? (
+                    <EuiLoadingSpinner />
+                  ) : (
+                    <ExpandableContent values={metadataValue.value ?? NOT_AVAILABLE_LABEL} />
+                  )}
+                </EuiDescriptionListDescription>
+              </EuiDescriptionList>
+            </EuiFlexItem>
+          ))}
+        </EuiFlexGroup>
+        <EuiFlexItem grow={false} key="metadata-link">
+          <EuiButtonEmpty
+            data-test-subj="infraMetadataSummaryShowAllMetadataButton"
+            onClick={onClick}
+            size="s"
+            flush="both"
+            iconSide="right"
+            iconType="sortRight"
+          >
+            <FormattedMessage
+              id="xpack.infra.assetDetailsEmbeddable.metadataSummary.showAllMetadataButton"
+              defaultMessage="Show all"
+            />
+          </EuiButtonEmpty>
```

Comment:
Hmm I see the button is also smaller there, it's hard to align it as the empty button has this blue background on click. With a smaller size looks a bit better and closer to the mockup I think (I made it larger as I thought it will be more visible but most of the links are smaller so I will keep it consistent): 
![image](https://github.com/elastic/kibana/assets/14139027/02aa7ada-0d5a-4e88-bc12-ccef2158a077)